presto-main/src/test/java/com/facebook/presto/cost/TestCoefficientBasedStatsCalculator.java

```diff
@@ -56,26 +55,24 @@ public void testStatsCalculatorUsesLayout()
     {
         assertPlan("SELECT orderstatus FROM orders WHERE orderstatus = 'P'",
                 anyTree(
-                        node(FilterNode.class,
-                                node(TableScanNode.class)
-                                        .withStats(PlanNodeStatsEstimate.builder()
-                                                .setOutputRowCount(new Estimate(363.0))
-                                                .setOutputSizeInBytes(unknownValue())
-                                                .build()))));
+                        node(TableScanNode.class)
+                                .withStats(PlanNodeStatsEstimate.builder()
+                                        .setOutputRowCount(new Estimate(363.0))
+                                        .setOutputSizeInBytes(unknownValue())
+                                        .build())));
 
         assertPlan("SELECT orderstatus FROM orders WHERE orderkey = 42",
                 anyTree(
-                        node(FilterNode.class,
-                                node(TableScanNode.class)
-                                        .withStats(PlanNodeStatsEstimate.builder()
-                                                .setOutputRowCount(new Estimate(15000.0))
-                                                .setOutputSizeInBytes(unknownValue())
-                                                .build()))));
+                        node(TableScanNode.class)
+                                .withStats(PlanNodeStatsEstimate.builder()
+                                        .setOutputRowCount(new Estimate(15000.0))
+                                        .setOutputSizeInBytes(unknownValue())
+                                        .build())));
     }
 
     private void assertPlan(String sql, PlanMatchPattern pattern)
     {
-        assertPlan(sql, LogicalPlanner.Stage.CREATED, pattern);
+        assertPlan(sql, LogicalPlanner.Stage.OPTIMIZED_AND_VALIDATED, pattern);
```

Comment:
it looks unrelated

Reply:
I think it is. We must ensure predicate is pushed down to TS node.